### PR TITLE
#2914 lastRequisitionDate incorrect [Branched from: #2909]

### LINKED ISSUE #2914: Item.LastRequisitionDate is incorrect
## Describe the bug

`Item.LastRequisitionDate` is incorrectly using `confirmDate` rather than `entryDate`

### To reproduce

Internal code bug

### Expected behaviour

As above

### Proposed Solution

As above

### Version and device info

- App version: Vaccine development 
- Tablet model: N/A
- OS version: N/A

### Additional context

N/A

### PR DESCRIPTION
**Pull request by joshxg**
Fixes #2914 

## Change summary

- Updates the `confirmDate` -> `entryDate`
- Off-road vial wastage fixes

## Testing

N/A

### Related areas to think about

N/A

### CODE REVIEW

File: src/database/DataTypes/Item.js [hunk 1]
```diff
@@ -283,10 +283,11 @@ export class Item extends Realm.Object {
     if (!this.isVaccine || !fromDate) return 0;
 
     const supplierCreditTransactionItems = this.transactionItems.filtered(
-      'transaction.confirmDate > $0 && transaction.type == $1 && transaction.otherParty.name == $2',
+      'transaction.confirmDate >= $0 && transaction.type == $1 &&' +
+        'transaction.otherParty.code == $2',
       fromDate,
       'supplier_credit',
-      'inventory_adjustment'
+      'invad'
```
Comment (joshxg) [inline src/database/DataTypes/Item.js:290]: off-road fix - invad seems to have funny namings - more consistent to use the code in mobile

File: src/database/DataTypes/Item.js [hunk 2]
```diff
@@ -256,7 +256,7 @@ export class Item extends Realm.Object {
     if (!this.isVaccine || !fromDate) return 0;
 
     const customerInvoiceTransactionItems = this.transactionItems.filtered(
-      "transaction.confirmDate > $0 && transaction.type == 'customer_invoice'",
+      "transaction.confirmDate >= $0 && transaction.type == 'customer_invoice'",
```
Comment (joshxg) [inline src/database/DataTypes/Item.js:259]: offroad..!

File: src/database/DataTypes/Item.js [hunk 3]
```diff
@@ -176,7 +176,7 @@ export class Item extends Realm.Object {
       .filtered("requisition.status == 'finalised'")
       .sorted('requisition.entryDate', true)[0];
 
-    return mostRecentRequisitionItem?.requisition?.confirmDate;
+    return mostRecentRequisitionItem?.requisition?.entryDate;
```
Comment (joshxg) [inline src/database/DataTypes/Item.js:179]: I'm stupid:(

Comment (mark-prins) [reply src/database/DataTypes/Item.js:179]: ah, I would have said `confirmDate` too. Probably did, in the PR review. I see it now.

**Pull request merged**